packages/p2p-v2/src/pages/my-ads/components/AdFormInput/__tests__/AdFormInput.spec.tsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import AdFormInput from '../AdFormInput';
+
+jest.mock('react-hook-form', () => ({
+    ...jest.requireActual('react-hook-form'),
+    Controller: ({ control, defaultValue, name, render }) =>
+        render({
+            field: { control, name, onBlur: jest.fn(), onChange: jest.fn(), value: defaultValue },
+            fieldState: { error: null },
+        }),
+    useFormContext: () => ({
+        control: 'mockedControl',
+        getValues: jest.fn(),
+    }),
+}));
+
+const mockProps = {
+    currency: 'usd',
+    label: 'label',
+    name: 'name',
+};
+
+describe('AdFormInput', () => {
+    it('should render the form input component', () => {
+        render(<AdFormInput {...mockProps} />);
+        expect(screen.getByText('label')).toBeInTheDocument();
+    });
+    it('should handle the input change', () => {
+        render(<AdFormInput {...mockProps} />);
+        const input = screen.getByRole('textbox');
```

Comment:
There might be a flow in this. What if in the future, there's more than one of these components with a role of `textbox `? This testcase might fail. isn't there another attribute we can combine with role to target the element? 🤔 
```suggestion
        const input = screen.getByRole('textbox');
```